custodial-copy-re-indexer/src/test/scala/uk/gov/nationalarchives/reindexer/MainSpec.scala

```diff
@@ -0,0 +1,64 @@
+package uk.gov.nationalarchives.reindexer
+
+import cats.effect.unsafe.implicits.global
+import cats.effect.{ExitCode, IO}
+import com.monovore.decline.effect.CommandIOApp
+import com.monovore.decline.{Command, Opts}
+import org.scalatest.EitherValues
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers.*
+import uk.gov.nationalarchives.reindexer.Arguments.index
+
+class MainSpec extends AnyFlatSpec with EitherValues:
+  object MainTest extends CommandIOApp(name = "test-app", header = ""):
+    override def main: Opts[IO[ExitCode]] = index.map(_ => IO(ExitCode.Success))
+
+  def runInvalidMainTest(args: String*): List[String] =
+    Command("", "")(MainTest.main)
+      .parse(args)
+      .left
+      .value
+      .errors
+
+  "Main" should "return success if the arguments are valid" in {
```

Comment:
All of the other tests use `IO` as its `--file-type` so we can't confirm for sure that `CO` is valid; do you think it's worth either duplicating this test and using `CO` (so that an error can be thrown in case `CO` becomes invalid somehow) or adding `CO` to some of the tests below which would potentially, in addition to throwing the other expected errors, will throw one if CO is/becomes invalid in the future?

Reply:
Yeah good point. I've put a table in for the two options and put a forAll around this test.